src/ol/interaction/drawboxinteraction.js

```diff
@@ -0,0 +1,360 @@
+goog.provide('ol.interaction.DrawBox');
+
+goog.require('goog.asserts');
+goog.require('goog.events');
+goog.require('ol.Collection');
+goog.require('ol.DrawEvent');
+goog.require('ol.Feature');
+goog.require('ol.FeatureOverlay');
+goog.require('ol.Map');
+goog.require('ol.MapBrowserEvent.EventType');
+goog.require('ol.Object');
+goog.require('ol.events.condition');
+goog.require('ol.geom.LineString');
+goog.require('ol.geom.Point');
+goog.require('ol.geom.Polygon');
+goog.require('ol.interaction.InteractionProperty');
+goog.require('ol.interaction.Pointer');
+
+
+
+/**
+ * @classdesc
+ * Interaction that allows drawing rectangular geometries. Supports using an
+ * {@link ol.events.condition}, shift key by default, to restrict the shape to
+ * a square.
+ *
+ * @constructor
+ * @extends {ol.interaction.Pointer}
+ * @fires ol.DrawEvent
+ * @param {olx.interaction.DrawBoxOptions} options Options.
+ * @api stable
+ */
+ol.interaction.DrawBox = function(options) {
+
+  goog.base(this, {
+    handleDownEvent: ol.interaction.DrawBox.handleDownEvent_,
+    handleEvent: ol.interaction.DrawBox.handleEvent,
+    handleUpEvent: ol.interaction.DrawBox.handleUpEvent_
+  });
+
+  /**
+     * Target source for drawn features.
+     * @type {ol.source.Vector}
+     * @private
+     */
```

Comment:
Minor, but to be consistent with the formatting of other comment blocks, this should be
```js
/**
 * Target source for drawn features.
 * @type {ol.source.Vector}
 * @private
 */
```